app/validation/validator.py

```diff
@@ -881,14 +922,36 @@ def _get_answers_with_parent_ids(json_to_validate):
         for section in json_to_validate.get('sections'):
             for group in section.get('groups'):
                 for block in group.get('blocks'):
-                    if block.get('questions'):
-                        for question in block['questions']:
-                            for answer in question.get('answers', []):
-                                answers[answer['id']] = {
-                                    'answer': answer,
-                                    'block': block['id'],
-                                    'group_id': group['id'],
-                                    'section': section['id']
-                                }
+                    for variant in block.get('variants', []):
+                        for answer in variant.get('question', {}).get('answers', []):
```

Comment:
Is it worth combining variants, questions and question in one place to avoid some dupe below, as long as it's still readable? 🤷‍♂️

Reply:
I flip flopped on this one. It's going to be changed fairly soon when `questions` is removed. But yeah, can probably be simpler. I'll change it.